### PR TITLE
Onboarding: Fix error if product types are missing

### LINKED ISSUE #3330: Onboarding: Skipping the product types step results in an error
**Describe the bug**
This is an edge case, but a quick fix.

If for some reason the product types step was skipped (e.g., a link was given to the final step or a third party plugin has filtered out this step), `profileItems.productTypes` is not defined in the final step.

We should check if this is defined before attempting to iterate over it.

**To Reproduce**

1. Delete `wc_onboarding_profile`.
2. Go to the profiler and skip to the last step `wp-admin/admin.php?page=wc-admin&step=theme`
3. Choose the currently active theme.
4. Note the console error.

**Expected behavior**
Expected this step to work independently of other steps.

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #3330

* Fixes an error that occurs if the product types step was somehow skipped. 
* Also removes duplicated efforts to redirect to the cart if purchasable items exist.

### Detailed test instructions:

1. Delete `wc_onboarding_profile` from `wp_options`.
2. Go to the profiler/dashboard.
3. Skip to the final step without completing the others using this URL: `wp-admin/admin.php?page=wc-admin&step=theme`
4. Choose the currently installed theme.
5. Make sure no error occurs and you see the task list.
6. Delete `wc_onboarding_profile` from `wp_options`.
7. Walk through the profiler, selecting product types and/or a theme that requires purchase.
8. Make sure you’re redirected to the cart as expected.

### CODE REVIEW

File: client/dashboard/profile-wizard/steps/theme/index.js [hunk 1]
```diff
@@ -78,48 +63,6 @@ class Theme extends Component {
...
-		const productIds = this.getProductIds();
-		const backUrl = getAdminLink( getNewPath( {}, '/', {} ) );
-		const { connectNonce } = getSetting( 'onboarding', {} );
-
-		const url = addQueryArgs( 'https://woocommerce.com/cart', {
-			'wccom-site': getSetting( 'siteUrl' ),
-			'wccom-woo-version': getSetting( 'wcVersion' ),
-			'wccom-back': backUrl,
-			'wccom-replace-with': productIds.join( ',' ),
-			'wccom-connect-nonce': connectNonce,
```
Comment (justinshreve) [inline client/dashboard/profile-wizard/steps/theme/index.js]: Oops, I didn't realize that we had competing code loaded here and in the dashboard.

The dashboard implementation seems to be missing a few of these parameters needed for the in-app purchase flow:

https://github.com/woocommerce/woocommerce-admin/blob/cc3849d05879ad08efa4b1e6817e27f57f732b3b/client/dashboard/index.js#L72-L75

Comment (joshuatf) [reply client/dashboard/profile-wizard/steps/theme/index.js]: >Oops, I didn't realize that we had competing code loaded here and in the dashboard.

Neither did I 😄 

>The dashboard implementation seems to be missing a few of these parameters needed for the in-app purchase flow:

Added these back in https://github.com/woocommerce/woocommerce-admin/pull/3340/commits/e83b8b72b0c578c3d3eb6db5c6fd7043e60f2612

**Comment by joshuatf** [comment]
Addressed your feedback in this PR @justinshreve and ready for another look.

File: client/dashboard/index.js [hunk 2]
```diff
@@ -62,15 +63,21 @@ class Dashboard extends Component {
...
 
 		document.body.classList.add( 'woocommerce-admin-is-loading' );
 
 		const url = addQueryArgs( 'https://woocommerce.com/cart', {
-			'wccom-back': window.location.href,
+			'wccom-site': getSetting( 'siteUrl' ),
+			'wccom-woo-version': getSetting( 'wcVersion' ),
 			'wccom-replace-with': productIds.join( ',' ),
+			'wccom-connect-nonce': connectNonce,
+			'wccom-back': backUrl,
```
Comment (justinshreve) [inline client/dashboard/index.js]: Even though this is what we had before, I seem to be having issues with `backUrl` containing the protocol and site url. The redirect ends up breaking on the WooCommerce.com side. Replacing this with just the `path` seems to fix it.

Comment (joshuatf) [reply client/dashboard/index.js]: Updated this to just the path.  Can you check and let me know if that fixes it on your end, @justinshreve?

I'm getting an `updateProfileItems()` loop at the end of the theme step on a Jurassic Ninja site the second time going through the profiler, but not sure if that's related to this PR or not.

**Pull request merged**